appengine/java/com/google/auth/appengine/AppEngineCredentials.java

```diff
@@ -51,7 +52,7 @@
  *
  * <p>Fetches access tokens from the App Identity service.
  */
-public class AppEngineCredentials extends GoogleCredentials {
+public class AppEngineCredentials extends GoogleCredentials implements ServiceAccountSigner {
```

Comment:
While I see why you are suggesting to move the `sign` method to the `GoogleCredentials` class I believe having a dedicated interface has some added value. Mainly because with a dedicated interface we can expose methods that take a `ServiceAccountSigner` and are granted that the provided parameter will support signing. Otherwise we would end up throwing runtime exceptions.

For instance in `google-cloud-java` we have the `Storage.signUrl` method that can take an option `SignUrlOption.signWith(ServiceAccountSigner)` to specify which account/credentials must be used to sign the URL.